src/arrayMethodSort.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFn =
+      compareFunction ||
+      function (a, b) {
+        if (a.toString() > b.toString()) {
+          return 1;
+        }
+
+        if (a.toString() < b.toString()) {
+          return -1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFn(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
You've implemented a bubble sort algorithm, which is perfectly acceptable for this task. However, bubble sort has a time complexity of O(n^2), which makes it inefficient for large arrays. If performance is a concern, consider researching and implementing more efficient sorting algorithms like quicksort or mergesort in the future.